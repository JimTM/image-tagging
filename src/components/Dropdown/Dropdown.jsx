import React from 'react';

const Dropdown = ({ tags, onSelectTag }) => {
  return (
    <div>
      <select defaultValue={'SELECTTAG'} onChange={e => onSelectTag(e)}>
        <option value="SELECTTAG">Select a tag</option>
        {tags.map(tag => (
          <option key={tag.id} value={tag.id}>
            {tag.label}
          </option>
        ))}
      </select>
    </div>
  );
};

export default Dropdown;
