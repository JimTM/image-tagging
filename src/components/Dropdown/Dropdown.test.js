import React from 'react';
import { shallow } from 'enzyme';

import Dropdown from './Dropdown';

describe('Dropdown tags component', () => {
  let wrapper = null;
  const tags = [{ id: 1, label: 'label 1' }, { id: 2, label: 'label 2' }];

  beforeEach(() => {
    wrapper = shallow(<Dropdown tags={tags} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct number of options', () => {
    const select = wrapper.find('select').first();
    expect(select.props().children.length).toEqual(tags.length);
  });
});
