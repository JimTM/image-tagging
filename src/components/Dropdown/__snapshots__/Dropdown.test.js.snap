// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown tags component should render 1`] = `ShallowWrapper {}`;
